src/main/java/com/zs/bodmas/Bodmas.java

```diff
@@ -23,21 +31,26 @@ public int evaluate(String expression) {
                 operands.push(expression.charAt(i));
             else if (expression.charAt(i) == ')') {
                 while (operands.peek() != '(')
-                    numbers.push(applyOp(operands.pop(), numbers.pop(), numbers.pop()));
+                    numbers.push(applyOperation(operands.pop(), numbers.pop(), numbers.pop()));
                 operands.pop();
             } else if (expression.charAt(i) == '+' || expression.charAt(i) == '-' ||
                     expression.charAt(i) == '*' || expression.charAt(i) == '/') {
-                while (!operands.empty() && hasPrecedence(expression.charAt(i), operands.peek()))
-                    numbers.push(applyOp(operands.pop(), numbers.pop(), numbers.pop()));
+                while (!operands.isEmpty() && hasPrecedence(expression.charAt(i), operands.peek()))
+                    numbers.push(applyOperation(operands.pop(), numbers.pop(), numbers.pop()));
                 operands.push(expression.charAt(i));
             }
         }
-        while (!operands.empty())
-            numbers.push(applyOp(operands.pop(), numbers.pop(), numbers.pop()));
+        while (!operands.isEmpty())
+            numbers.push(applyOperation(operands.pop(), numbers.pop(), numbers.pop()));
```

Comment:
your program is not giving a correct output. Please debug this and find what is going wrong and discuss with me.
Also take care of the boundary conditions like divide by zero and what if you get some unknown character in you string.